x-pack/plugins/asset_manager/docs/api.md

```diff
@@ -0,0 +1,187 @@
+# Asset Manager API Documentation
+
+## Plugin configuration
+
+This plugin is NOT fully enabled by default, even though it's always enabled
+by Kibana's definition of "enabled". However, without the following configuration,
+it will bail before it sets up any routes or returns anything from its
+start, setup, or stop hooks.
+
+To fully enable the plugin, set the following config values in your kibana.yml file:
+
+```yaml
+xpack.assetManager:
+  alphaEnabled: true
+```
+
+## Depending on an asset client in your packages
+
+If you're creating a shared UI component or tool that needs to access asset data, you
+can create that code in a stateless Kibana package that can itself be imported into
+any Kibana plugin without any dependency restrictions. To gain access to the asset data,
+this component or tool can require the appropriate asset client to be passed in.
+
+TODO: need to move main client types to a package so that they can be depended on by
```

Comment:
can the packages import types from assetmanager plugin instead ?

Reply:
Should packages import from plugins? That sounds wrong, but maybe it's allowed for types only? Not sure it's a good idea...

Reply:
packages already do that (eg [kbn-data-service](https://github.com/elastic/kibana/blob/main/packages/kbn-data-service/tsconfig.json#L10), [alert_details](https://github.com/elastic/kibana/blob/main/x-pack/packages/observability/alert_details/tsconfig.json#L23)..) but I don't know if that's recommended

Reply:
Ok, I'll try to poke around and think on that a bit.